src/main/java/org/killbill/billing/plugin/util/http/HttpClient.java

```diff
@@ -139,24 +188,71 @@ public Response onCompleted(final Response response) throws Exception {
             throw new InvalidRequest("Invalid request", response);
         }
 
-        return deserializeResponse(response, clazz);
+        return deserializeResponse(response, clazz, format);
     }
 
-    protected <T> T deserializeResponse(final Response response, final Class<T> clazz) throws IOException {
+    protected <T> T deserializeResponse(final Response response, final Class<T> clazz, ResponseFormat format) throws IOException {
         InputStream in = null;
         try {
             in = response.getResponseBodyAsStream();
-            return mapper.readValue(in, clazz);
+            switch (format) {
+                case TEXT:
+                    return (T) CharStreams.toString(new InputStreamReader(in, Charsets.UTF_8));
+                default:
+                    return mapper.readValue(in, clazz);
+            }
         } finally {
             if (in != null) {
                 in.close();
             }
         }
     }
 
+    @Deprecated
```

Comment:
👍 for keeping existing methods.

Could you also double check it still works as expected (see `TestRemoteAdyenPaymentServiceProviderHostedPaymentPagePort.java` in Adyen for instance).